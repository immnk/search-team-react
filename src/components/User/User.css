.root {
  border: 1px solid #cbcddc80;
  box-shadow: 0 1px 3px #aaaaaa1f, 0 1px 2px #aaaaaa3d;
}

.figure {
  text-align: center;
  margin: 1.2rem;

  img {
    width: 100px;
    height: 100px;
    border-radius: 50%;
    border: 3px solid #fff;
    box-shadow: 0 1px 3px #aaaaaa1f, 0 1px 2px #aaaaaa3d;
  }
}

.icon {
}

.details {
  padding: 1.2rem;

  h3 {
    font-size: 2rem;
    text-align: center;
    font-weight: 400;
  }

  a {
    display: flex;
    align-items: center;
    text-decoration: none;
    margin-bottom: 0.5rem;

    .icon {
      color: #4d4f59b3;
      margin-right: 0.5rem;
    }
  }
}

.username {
  color: grey;
  font-size: 1.2rem;
  text-transform: lowercase;
  font-weight: 100;
  text-align: center;
  margin-bottom: 1.2rem;
}
