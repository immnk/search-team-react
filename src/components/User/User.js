import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './User.css';

const User = ({ className, user }) => {
  const { id, name, username, email, address, phone } = user;
  return (
    <section className={classNames(css.root, className)}>
      <figure className={css.figure}>
        <img
          src={`https://source.unsplash.com/collection/888146/300x300?id=${id}&name=${name}`}
          alt={name}
        />
        {/* <img src={`https://randomuser.me/api/portraits/med/women/${id}.jpg`} alt={name} /> */}
      </figure>
      <aside className={css.details}>
        <h3>{name}</h3>
        <p className={css.username}>{username}</p>
        <a href={`mailto:${email}`}>
          <span className={classNames('material-icons', css.icon)}>contact_mail</span>
          <span>{email}</span>
        </a>
        <a href={`tel:${phone}`}>
          <span className={classNames('material-icons', css.icon)}>call</span>
          <span>{phone}</span>
        </a>
        <a
          href={`http://www.google.com/maps/place/${address.geo.lat},${address.geo.lng}`}
          target="_blank"
          rel="noreferrer"
        >
          <span className={classNames('material-icons', css.icon)}>place</span>
          <span>Locate Me</span>
        </a>
      </aside>
    </section>
  );
};

User.defaultProps = {
  className: undefined,
  user: {}
};

User.propTypes = {
  className: PropTypes.string,
  user: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    username: PropTypes.string,
    email: PropTypes.string,
    phone: PropTypes.string,
    address: PropTypes.shape({
      geo: PropTypes.shape({
        lat: PropTypes.string,
        lng: PropTypes.string
      })
    })
  })
};

export default User;
