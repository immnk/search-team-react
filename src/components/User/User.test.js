import React from 'react';
import { shallow } from 'enzyme';

import User from './User';

const renderUser = (props) => {
  const defaultProps = {
    className: 'my-classname',
    user: {
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz',
      address: {
        street: 'Kulas Light',
        suite: 'Apt. 556',
        city: 'Gwenborough',
        zipcode: '92998-3874',
        geo: {
          lat: '-37.3159',
          lng: '81.1496'
        }
      },
      phone: '1-770-736-8031 x56442',
      website: 'hildegard.org',
      company: {
        name: 'Romaguera-Crona',
        catchPhrase: 'Multi-layered client-server neural-net',
        bs: 'harness real-time e-markets'
      }
    }
  };

  return <User {...defaultProps} {...props} />;
};

const shallowRenderUser = (props) => shallow(renderUser(props));

describe('Test User', () => {
  it('render app', () => {
    const actual = shallowRenderUser();
    expect(actual).toMatchSnapshot();
  });
});
