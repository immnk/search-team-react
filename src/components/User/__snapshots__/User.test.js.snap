// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test User render app 1`] = `
<section
  className="root my-classname"
>
  <figure
    className="figure"
  >
    <img
      alt="Leanne Graham"
      src="https://source.unsplash.com/collection/888146/300x300?id=1&name=Leanne Graham"
    />
  </figure>
  <aside
    className="details"
  >
    <h3>
      Leanne Graham
    </h3>
    <p
      className="username"
    >
      Bret
    </p>
    <a
      href="mailto:Sincere@april.biz"
    >
      <span
        className="material-icons icon"
      >
        contact_mail
      </span>
      <span>
        Sincere@april.biz
      </span>
    </a>
    <a
      href="tel:1-770-736-8031 x56442"
    >
      <span
        className="material-icons icon"
      >
        call
      </span>
      <span>
        1-770-736-8031 x56442
      </span>
    </a>
    <a
      href="http://www.google.com/maps/place/-37.3159,81.1496"
      rel="noreferrer"
      target="_blank"
    >
      <span
        className="material-icons icon"
      >
        place
      </span>
      <span>
        Locate Me
      </span>
    </a>
  </aside>
</section>
`;
